src/api/FileCollaborators.js

```diff
@@ -24,26 +30,34 @@ class FileCollaborators extends MarkerBasedAPI {
     }
 
     /**
-     * Generic success handler
+     * Transform result of API response
      *
      * @param {Object} data the response data
      */
-    successHandler = (data: any): void => {
+    successHandler = (data: CollaboratorsAPIResponse): void => {
         if (this.isDestroyed() || typeof this.successCallback !== 'function') {
             return;
         }
 
-        const { entries } = data;
-        const collaborators = entries.map(collab => {
-            const { id, name, login } = collab;
-            return {
-                id,
-                name,
-                item: { id, name, email: login },
-            };
-        });
+        // Transform into "mention selector" format:
+        const collaboratorSelectorItems: SelectorItems<UserMini | GroupMini> = data.entries.map(
+            (collab: UserMini | GroupMini) => {
+                let item;
+                if (collab.type === 'group') {
+                    item = collab; // flow needs assignment to happen after type refinement
+                } else {
+                    item = collab;
+                    item.email = item.login; // transform user object
```

Comment:
Difference: this leaves "login" in the user object as well as aliasing it to "email"; may be useful if we move more towards using the API response format without transforming this field everywhere.